src/modules/commander/Commander.cpp

```diff
@@ -1832,78 +1831,80 @@ Commander::run()
 		}
 
 		// Geofence actions
-		if (armed.armed && (geofence_result.geofence_action != geofence_result_s::GF_ACTION_NONE)) {
+		// Geofence failsafe is disabled when in a low battery handling action (critical or emergency)
+		if (armed.armed
+		    && (geofence_result.geofence_action != geofence_result_s::GF_ACTION_NONE)
+		    && (_battery_warning < battery_status_s::BATTERY_WARNING_CRITICAL)) {
```

Comment:
The fact that you need to make a comment to describe the condition means to me that it should be broken out in order to explain it:

```
const bool geofence_action_enabled = geofence_result.geofence_action != geofence_result_s::GF_ACTION_NONE;
const bool battery_critical_or_emergency = _battery_warning < battery_status_s::BATTERY_WARNING_CRITICAL;

if (...
```